### PR TITLE
Add Hooks for Liquid-Liquid Point Model 

### PR DESCRIPTION
**Pull request by ggrbill**
Also, Fix some internal links in the documentation, add missing docs (TwoPhaseSystem) and remove unused enumeration (WallLayerProperty).

### CODE REVIEW

**Review by tarcisiofischer — APPROVED** [review]

File: docs/source/plugins/06_solver_hooks.rst [hunk 1]
```diff
@@ -185,6 +185,53 @@ flow is in the control volume.
     Another important API function for UCM is :cpp:func:`get_ucm_fluid_geometrical_properties`. This function computes
     the geometrical properties properly in each previous presented `hooks` depending on the flow pattern.
 
+Unit Cell Model (UCM) Liquid-Liquid System
```
[the remark this answers is not in the document]
Comment (ggrbill) [reply docs/source/plugins/06_solver_hooks.rst:188]: As we discussed in a call, we decide to change names from Unit Cell Model (UCM) to Mechanistic Model. With that, the Hooks and API functions related to Liquid-Liquid System will lost the `_ucm` from their names. Internally, in ALFAsim, we decided to keep as few changes as possible.

What do you think, @arthursoprana ?

File: src/alfasim_sdk/alfasim_sdk_api/api.h [hunk 2]
```diff
@@ -633,4 +634,123 @@ DLL_EXPORT int get_ucm_fluid_geometrical_properties(
...
+    its documentation should be visited for more details.
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] out Variable value.
+    @param[in] var_name String with the variable name. See the list of possible values above.
+    @param[in] phase_id A #LiquidLiquidSystem value. When the requested variable is not associated
+                            to a phase any value can be passed.
+    @return An #error_code value.
+*/
+DLL_EXPORT int get_ucm_liqliq_shear_force_per_volume_input_variable(
```
Comment (arthursoprana) [inline src/alfasim_sdk/alfasim_sdk_api/api.h:752]: Minor, but I think we should keep undescores in `liqliq` -> `liq_liq`.

This API function will be renamed to `get_liq_liq_shear_force_per_volume_input_variable`, right?

File: docs/source/plugins/06_solver_hooks.rst [hunk 3]
```diff
@@ -185,6 +185,53 @@ flow is in the control volume.
...
+------------------------------------------
+
+The |alfasim|'s multiphase modeling represents an one dimensional multiphase flow, in which is a result of some
```
Comment (arthursoprana) [inline docs/source/plugins/06_solver_hooks.rst:191]:
```suggestion
|alfasim|'s multiphase model assumes a one dimensional system of nonlinear equations, in which is a result of some
```

File: docs/source/plugins/06_solver_hooks.rst [hunk 4]
```diff
@@ -185,6 +185,53 @@ flow is in the control volume.
...
+simplifications that transforms a local instantaneous three-dimensional model into a one-dimensional model. It results
+in a set of average equations for the multiphase flow. Because of those simplifications some important phenomena must
+be modeled in order to accurately represent the physical behavior. The shear stress is one of these phenomena that is
+modeled to increase the quality of the one dimensional model, in which the Unit Cell model (UCM) is used.
```
Comment (arthursoprana) [inline docs/source/plugins/06_solver_hooks.rst:195]:
```suggestion
|alfasim|'s resulting multiphase model is a one-dimensional system of nonlinear equations, which is a result of some
averaging steps that transforms a local instantaneous three-dimensional model into a one-dimensional multiphase (and multi-field) model. The derived variables that form the nonlinear differential equations represent average values at a given point in time and space. It is clear that in this averaging process there are important phenomena that will non longer be naturally captured from the equations and therefore one of the strategies is to incorporate all the physics that have been lost in the averagig process in the shear stress term in order to accurately represent the physical behavior. Hence the shear stress is one of the most important parameters for characterizing the quality of a one-dimensional multiphase model.
```

File: docs/source/plugins/06_solver_hooks.rst [hunk 5]
```diff
@@ -185,6 +185,53 @@ flow is in the control volume.
...
+
+.. note ::
+    Note that the two phase (Gas-Liquid) system of UCM can be used for both two and three phase flows, in which is
+    done in case of a liquid-liquid system is not available via plugin.
+
+In the gas-liquid system the liquid phase is considered as a sum of all liquid phases, and the model take into account
+only the interaction behavior between these two phases (Gas and Liquid). Of course, this kind of approach brings some
+limitations, since the behavior of the oil and water in the liquid phase is considered nearly homogeneous (mixture).
+On the other hand, if the liquid-liquid system considers the interaction between oil and water, it introduces a better
+modeling and representation of the tree phase system. In fact, considering the interaction between the oil and water
```
Comment (arthursoprana) [inline docs/source/plugins/06_solver_hooks.rst:210]:
```suggestion
modeling and representation of the three-phase system. In fact, considering the interaction between the oil and water
```

File: docs/source/plugins/06_solver_hooks.rst [hunk 6]
```diff
@@ -185,6 +185,53 @@ flow is in the control volume.
...
+phases, behaviors such as emulsion formation (oil dominated or water dominated) and stratified flow can be taken into
+account together with the gas-liquid system.
+
+The |alfasim|'s Unit Cell Model (UCM) of three phase system is divided into two steps, one for gas-liquid system and
+one for liquid-liquid system in which the complete representation uses an incremental approach. The two phase systems
+(Gas-Liquid and Liquid-Liquid) model different behaviors and are coupled in the three phase system algorithm. First the
+UCM for gas-liquid system is solved, considering a single liquid phase (Oil and Water mixture), after that the UCM for
+liquid-liquid system is solved, considering independent oil and water phases. Before assume that both have a final
+result, the properties of liquid phase are updated taking into account the solution of UCM for liquid-liquid system.
```
Comment (arthursoprana) [inline docs/source/plugins/06_solver_hooks.rst:219]:
```suggestion
liquid-liquid system is solved, considering independent oil and water phases. Finally, the properties of liquid phase are updated taking into account the solution of UCM for liquid-liquid system.
```

File: src/alfasim_sdk/alfasim_sdk_api/api.h [hunk 7]
```diff
@@ -633,4 +634,123 @@ DLL_EXPORT int get_ucm_fluid_geometrical_properties(
...
+    - `"mu"`: Viscosity [Pa.s]
+    - `"U_S"`: Superficial Velocity [m/s]
+
+    It is important to know that the listed `variable_name`s are not available in any phase, only for
+    liquid-liquid systems, in which `OIL` (0, zero) and `WATER` (1, One) are the possible values, and
+    they are available in the #LiquidLiquidSystem enumeration.
+    Because of that, the #error_code must be checked.
+
+    List of `variable_name` with one value:
+    - `"alpha_w"` : Water Volume Fraction [kg of `water phase` / kg of `liquid(Oil+Water) phase`]
```
Comment (arthursoprana) [inline src/alfasim_sdk/alfasim_sdk_api/api.h:689]:
```suggestion
    - `"alpha_w"` : Water Volume Fraction [m3 of `water phase` / m3 of `liquid(Oil+Water) phase`]
```

File: src/alfasim_sdk/alfasim_sdk_api/api.h [hunk 8]
```diff
@@ -633,4 +634,123 @@ DLL_EXPORT int get_ucm_fluid_geometrical_properties(
...
+    - `"D_h"`: Unit Cell Liquid (Oil-Water) Hydraulic Diameter [m]
+    - `"ks"`: Roughness [m]
```
Comment (arthursoprana) [inline src/alfasim_sdk/alfasim_sdk_api/api.h:691]: Absolute roughness? Or relative?

**Review by arthursoprana — CHANGES_REQUESTED** [review]
Nice work, take a look at some of my comments.

File: src/alfasim_sdk/alfasim_sdk_api/api.h (hunk 2 above)
Comment (ggrbill) [reply src/alfasim_sdk/alfasim_sdk_api/api.h:752]: Yes, exactly. I think it is a good idea to put an underscore in the name.

File: docs/source/plugins/06_solver_hooks.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (ggrbill) [reply docs/source/plugins/06_solver_hooks.rst:188]: - [ ] remove `_ucm`

File: src/alfasim_sdk/alfasim_sdk_api/api.h (hunk 8 above)
Comment (ggrbill) [reply src/alfasim_sdk/alfasim_sdk_api/api.h:691]: Thanks, I forgot it. It is Absolute roughness.

- [ ] Absolute Roughness

**Comment by ggrbill** [comment]
> Nice work, take a look at some of my comments.

File: docs/source/plugins/06_solver_hooks.rst (hunk 4 above)
Comment (ggrbill) [reply docs/source/plugins/06_solver_hooks.rst:195]: Done

File: docs/source/plugins/06_solver_hooks.rst (hunk 5 above)
Comment (ggrbill) [reply docs/source/plugins/06_solver_hooks.rst:210]: Done

File: src/alfasim_sdk/alfasim_sdk_api/api.h (hunk 7 above)
Comment (ggrbill) [reply src/alfasim_sdk/alfasim_sdk_api/api.h:688]: Done

File: docs/source/plugins/06_solver_hooks.rst (hunk 6 above)
Comment (ggrbill) [reply docs/source/plugins/06_solver_hooks.rst:219]: Done

**Pull request merged**